src/rust/engine/options/src/lib.rs

```diff
@@ -225,6 +225,44 @@ impl Source {
     }
 }
 
+pub fn apply_list_edits<T>(
```

Comment:
This was refactored out of `parse_list` below, as we now want to use it on the intermediate values in the derivation, and not just to compute the final value.